functions/src/glean.ts

```diff
@@ -0,0 +1,228 @@
+import Glean from "@mozilla/glean/node";
+import PingEncryptionPlugin from "@mozilla/glean/plugins/encryption";
+import axios from "axios";
+import { Mutex, Semaphore, withTimeout } from "async-mutex";
+
+import * as rallyMetrics from "./generated/rally.js";
+import * as userMetrics from "./generated/user.js";
+import * as enrollmentMetrics from "./generated/enrollment.js";
+import * as unenrollmentMetrics from "./generated/unenrollment.js";
+import * as rallyPings from "./generated/pings.js";
+
+import {
+  Uploader,
+  UploadResult,
+  UploadResultStatus,
+} from "@mozilla/glean/uploader";
+
+const GLEAN_DEBUG_VIEW_TAG = "MozillaRally";
+const GLEAN_RALLY_APP_ID = process.env.NODE_ENV !== "test" ? "rally-core" : "test-app-id";
+const GLEAN_APP_DISPLAY_VERSION = "TODO_app_version";
```

Comment:
not sure where to take the version from

Reply:
I think the version of the rally-web-platform would make sense, [this is how we grab that](https://github.com/mozilla-rally/rally-web-platform/blob/87df3d4b87d7a0dd2188e75dc24e014609e85549/scripts/add-version-object.js#L4) for the the [version.json](stage.rally-web.nonprod.dataops.mozgcp.net/version.json) URL.